Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -198,8 +207,16 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
             logger.debug("Connection pool exhausted on this event loop, adding request to waitlist")
             let promise = eventLoop.makePromise(of: Source.Connection.self)
             self.waiters.append((logger, promise))
-            // return waiter
-            return promise.futureResult
+            
+            let task = eventLoop.scheduleTask(in: self.requestTimeout) { [weak self, logger, promise] in
+                logger.error("Connection request timed out. This might indicate a connection deadlock in your application.")
+                if let idx = self?.waiters.firstIndex(where: { let p = $0.1; return p.futureResult === promise.futureResult }) {
```

Comment:
I think closure labels and some whitespace here would be nice to avoid `let p = $0.1;` which is overly terse.